src/components/Feedback/index.tsx

```diff
@@ -0,0 +1,177 @@
+import { Amplify } from '@aws-amplify/core';
+import { API } from '@aws-amplify/api';
+import { useRef, useState } from 'react';
+import {
+  FeedbackContainer,
+  VoteButton,
+  VoteButtonsContainer,
+  Toggle,
+  FeedbackMobileContainer,
+  ThankYouContainer
+} from './styles';
+import awsconfig from '../../aws-exports';
+import { useEffect } from 'react';
+import isUUID from 'validator/lib/isUUID';
+import { trackFeedbackSubmission } from '../../utils/track';
+
+Amplify.configure(awsconfig);
+if (process.env.API_ENV === 'production') {
+  Amplify.configure({
+    aws_cloud_logic_custom: [
+      {
+        name: 'submissions',
+        endpoint: 'https://docs-backend.amplify.aws',
+        region: 'us-west-2'
+      }
+    ]
+  });
+}
+
+enum FeedbackState {
+  START = 'START',
+  END = 'END',
+  HIDDEN = 'HIDDEN'
+}
+
+type Feedback = {
+  vote: boolean;
+  page_path: string;
+  id?: string;
+  comment?: string;
+};
+
+export default function Feedback() {
+  const [state, setState] = useState<FeedbackState>(FeedbackState.START);
+  const [feedbackId, setFeedbackId] = useState(undefined);
+  const feedbackQuestion = 'Was this page helpful?';
+  const feedbackAppreciation = 'Thank you for your feedback!';
+
+  useEffect(() => {
+    // UUID of feedback if it exists.
+    const id = window.localStorage.getItem('feedbackId');
+    if (id && isUUID(id)) {
+      setFeedbackId(id);
+    }
+  }, []);
+
+  async function submitVote(vote: boolean, comment?: string) {
+    // Path without heading link
+    const pagePath = window.location.href.split('#')[0];
+
+    const body: Feedback = {
+      page_path: pagePath,
+      vote: vote
+    };
+
+    const headers = {
+      'content-type': 'application/json'
+    };
+
+    if (feedbackId) {
+      body.id = feedbackId;
+    }
+
+    try {
+      trackFeedbackSubmission(vote);
+
+      const result = await API.post('submissions', '/submissions', {
+        headers,
+        body
+      });
+
+      if (!feedbackId && result?.data) {
+        const data = JSON.parse(result.data);
+
+        if (data.id) {
+          window.localStorage.setItem('feedbackId', data.id);
+        }
+      }
+    } catch (err) {
+      console.log(err);
+    }
+  }
+
+  return (
+    <FeedbackContainer style={state === FeedbackState.HIDDEN ? {display: 'none'} : {} }>
+      {state == FeedbackState.START ? (
+        <>
+          <p>{feedbackQuestion}</p>
+          <VoteButtonsContainer>
+            <VoteButton
+              onClick={() => {
+                setState(FeedbackState.END);
+                submitVote(true);
```

Comment:
T.J. suggests to swap these in case submission fails because a user won't know if there's a success or not